securesignals-ima-devapp/src/main/java/com/uid2/dev/MainActivity.java

```diff
@@ -0,0 +1,283 @@
+
+        // Add listeners for when ads are loaded and for errors.
+        adsLoader.addAdErrorListener(
+            new AdErrorEvent.AdErrorListener() {
+                /** An event raised when there is an error loading or playing ads. */
+                @Override
+                public void onAdError(AdErrorEvent adErrorEvent) {
+                    Log.i(LOGTAG, "Ad Error: " + adErrorEvent.getError().getMessage());
+                    resumeContent();
+                }
+            });
+        adsLoader.addAdsLoadedListener(
+            new AdsLoader.AdsLoadedListener() {
+                @Override
+                public void onAdsManagerLoaded(AdsManagerLoadedEvent adsManagerLoadedEvent) {
+                    // Ads were successfully loaded, so get the AdsManager instance. AdsManager has
+                    // events for ad playback and errors.
+                    adsManager = adsManagerLoadedEvent.getAdsManager();
+
+                    // Attach event and error event listeners.
+                    adsManager.addAdErrorListener(
+                        new AdErrorEvent.AdErrorListener() {
+                            /** An event raised when there is an error loading or playing ads. */
+                            @Override
+                            public void onAdError(AdErrorEvent adErrorEvent) {
+                                Log.e(LOGTAG, "Ad Error: " + adErrorEvent.getError().getMessage());
+                                String universalAdIds =
+                                    Arrays.toString(adsManager.getCurrentAd().getUniversalAdIds());
+                                Log.i(
+                                    LOGTAG,
+                                    "Discarding the current ad break with universal "
+                                        + "ad Ids: "
+                                        + universalAdIds);
+                                adsManager.discardAdBreak();
+                            }
+                        });
+                    adsManager.addAdEventListener(
+                        new AdEvent.AdEventListener() {
+                            /** Responds to AdEvents. */
+                            @Override
+                            public void onAdEvent(AdEvent adEvent) {
+                                if (adEvent.getType() != AdEvent.AdEventType.AD_PROGRESS) {
+                                    Log.i(LOGTAG, "Event: " + adEvent.getType());
+                                }
+                                // These are the suggested event types to handle. For full list of
+                                // all ad event types, see AdEvent.AdEventType documentation.
+                                switch (adEvent.getType()) {
+                                    case LOADED:
+                                        // AdEventType.LOADED is fired when ads are ready to play.
+
+                                        // This sample app uses the sample tag
```

Comment:
Nit: I know this is copied from a different app, but they are using a different max line length than we have configured for this project. Consider reformatting these comment blocks to use all `120` characters.

Reply:
Done.